pkgs/development/python-modules/ansible/collections.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, ansible-base
+, jsonschema
+, jxmlease
+, ncclient
+, netaddr
+, paramiko
+, pynetbox
+, scp
+, textfsm
+, ttp
+, xmltodict
+, withJunos ? false
+, withNetbox ? false
+}:
+
+buildPythonPackage rec {
+  pname = "ansible";
+  version = "3.4.0";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "096rbgz730njk0pg8qnc27mmz110wqrw354ca9gasb7rqg0f4d6a";
+  };
+
+  postPatch = ''
+    # make ansible-base depend on ansible-collection, not the other way around
+    sed -i '/ansible-base/d' setup.py
+  '';
+
+  propagatedBuildInputs = lib.unique ([
+    # Support ansible collections by default, make all others optional
+    # ansible.netcommon
+    jxmlease
+    ncclient
+    netaddr
+    paramiko
+    xmltodict
+    # ansible.posix
+    # ansible.utils
+    jsonschema
+    textfsm
+    ttp
+    xmltodict
+    # ansible.windows
+
+    # lots of collections with dedicated requirements.txt and pyproject.toml files,
+    # add the dependencies for the collections you need conditionally and install
+    # ansible using overrides to enable the collections you need.
+  ] ++ lib.optionals (withJunos) [
+    # ansible_collections/junipernetworks/junos/requirements.txt
+    jxmlease
+    ncclient
+    paramiko
+    scp
+    xmltodict
+  ] ++ lib.optionals (withNetbox) [
+    pynetbox
+  ]);
+
+  dontStrip = true;
```

Comment:
why is this needed?

Reply:
There shouldn't be anything to strip in there and `find [...] -exec strip [...]` on close to 50k files adds roughly 6 minutes to the build process.

Reply:
oh, do you mind adding that as a comment? otherwise someone might try removing it

Reply:
Copying from home-assistant :smile: 

```
  # don't try and fail to strip 6600+ python files, it takes minutes!
  dontStrip = true;
```